spec/models/preserved_copy_spec.rb

```diff
@@ -194,4 +194,18 @@
       expect(preserved_copy.matches_po_current_version?).to be false
     end
   end
+
+  context '.least_recent_version_audit(last_checked_b4_date, storage_dir)' do
+    let(:db_query) {
+      described_class.least_recent_version_audit(Time.now.utc, 'spec/fixtures/storage_root01/moab_storage_trunk')
+    }
+
+    it 'returns an ActiveRecord::Relation' do
+      expect(db_query).to be_a(ActiveRecord::Relation)
+    end
+
+    it 'returns the preserved_copy in the db' do
```

Comment:
does it return one preserved_copy, or an array of them?

Reply:
confusingly, i think it's the test name that's wrong here.  the code currently uses this for the order by:
`'last_version_audit IS NOT NULL, last_version_audit ASC'`

what i think that does is sort nulls first, then non-nulls in ascending order.  the logic being:
* the first sort criterion is: 'date not null?'  all rows where the date is null will be false by that criterion, and i believe false results sort above true results (0 < 1).
* the second sort criterion just the date in ascending order, so all non-null dates will be sorted next by the actual date value.

in oracle, you could just do `'last_version_audit ASC NULLS FIRST'`, but when i went looking around on the internet to see if that was standard SQL, i saw stack exchange posts and whatnot jumping through hoops because they said it wasn't.  didn't do any testing of my own to confirm whether they were right.

Reply:
An array: ```#<ActiveRecord::Relation [#<PreservedCopy id: 1, version: 0, preserved_object_id: 1, endpoint_id: 1, created_at: "2018-01-29 22:04:52", updated_at: "2018-01-29 22:04:52", last_moab_validation: nil, last_checksum_validation: nil, size: 1, status: "validity_unknown", last_version_audit: nil>]>``` I can change the description to clear that up, if it's too ambiguous.

Reply:
Well, it returns one preserved copy as an AR::Relation that you can call `to_a` on to generate an array from, so actually, I'm not sure what to say.

Reply:
yeah, i think technically the scope (like a `join` or a `where` or whatever used directly) returns an ActiveRecord relation.  and i *think* AR relations have some magic to so that they behave somewhat transparently like the object or objects you likely want if you start using that relation directly as if it were the query result itself (i.e. as if the AR relation is actually the AR model instance).  which would mean the result has the class `ActiveRecord::Relation`, but that it might behave like a list, or like a single result, depending on how you interact with it.  ✨

Reply:
💀 🐔 💀 🐔 💀  <-- voodoo

🎩 🐰 <-- magic